qtile_builder.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2033, elParaguayo
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import os
+import subprocess
+import sys
+from pathlib import Path
+
+import setuptools.build_meta as sbm
+from setuptools.build_meta import LEGACY_EDITABLE
+
+ROOT = Path(__file__).parent.resolve()
+
+
+def build_wayland():
+    """
+    Calls the wayland build script and ensures the folder is in the
+    python path.
+    """
+    print("Building wayland backend cffi bindings...")
+    sys.path.insert(0, ROOT.as_posix())
+    python_path = ":".join(sys.path)
+    try:
+        subprocess.run(
+            [sys.executable, "libqtile/backend/wayland/cffi/build.py"],
+            check=True,
+            env=os.environ.update({"PYTHONPATH": python_path}),
```

Comment:
`os.environ.update({"PYTHONPATH": python_path})` returns `None`. You need could do:
```suggestion
            env=os.environ | {"PYTHONPATH": python_path},
```

Reply:
Interesting. Maybe this isn't needed then as it built ok for me locally. Need to test again.

Also, should we just have this run `ffibuild` instead that way we can just manage building in a single script?